GitTfs/Core/GitTfsRemote.cs

```diff
@@ -325,9 +325,12 @@ public IFetchResult FetchWithMerge(int mergeChangesetId, bool stopOnFailMergeCom
             var latestChangesetId = GetLatestChangesetId();
             if (lastChangesetIdToFetch != -1)
                 latestChangesetId = Math.Min(latestChangesetId, lastChangesetIdToFetch);
-            // TFS 2010 doesn't like when we ask for history past its last changeset.
+
             if (MaxChangesetId >= latestChangesetId)
+            {
+                Trace.WriteLine("TFS 2010 doesn't like when we ask for history past its last changeset.");
```

Comment:
I fear that this message won't be much more helpful than the comment was. Should `git-tfs` have been run differently? Did git-tfs get itself into a weird state?